File: structural/adapter/typescript/adapter.ts

import { OldPaymentSystem } from "./adaptee";
import { PaymentProcessor } from "./target";

export class PaymentAdapter implements PaymentProcessor {
    private oldSystem: OldPaymentSystem;

    constructor(oldSystem: OldPaymentSystem) {
        this.oldSystem = oldSystem;
    }

    makePayment(amount: number): string {
        return this.oldSystem.processPayment(amount);
    }
}

File: structural/adapter/python/adapter.py

from structural.adapter.python.adaptee import OldPaymentSystem
from structural.adapter.python.target import PaymentProcessor


class PaymentAdapter(PaymentProcessor):
    def __init__(self, old_system: OldPaymentSystem) -> None:
        self._old_system = old_system

    def make_payment(self, amount: float) -> str:
        return self._old_system.process_payment(amount)